aten/src/ATen/native/native_functions.yaml

```diff
@@ -104,6 +104,32 @@
 - func: _unpack_dual(Tensor(a) dual, int level) -> (Tensor(a) primal, Tensor tangent)
   variants: function
 
+# NOTE: [_new_zeros_with_same_feature_meta]
+# This function creates a new tensor with the layout and TensorOptions
+# of `other` but also takes into account the batch dimensions of `self`
+#
+# This function has a couple extra constraints because it is also used for `jvp`
+# in functorch.
+# - is used for forward AD because there is the restriction
+#   that the primal and tangent must have the same layout
+# - We cannot assume that `self` and `other` have the same sizes or even dim
+#   because in the inplace over view case, `other` is the base tensor, and
+#   `self` is the forward grad with respect to the view, which can have an
+#   entirely different shape
+# - takes the number of batch dims for `self` because we also handle
+#   some batching logic. We handle that here instead of a batching rule because
+#   we'd like to avoid calling as_strided in the batching rule (as to enable
+#   nested vmap in functorch).
+# - needs to be CompositeExplicitAutograd for jvp support in functorch.
+#   functorch currently relies on TensorWrapper which does not have storage
+#   CompositeExplicitAutograd makes sure the TensorWrapper is unwrapped.
+# - this function may eventually take on another int argument to store the
+#   the number of batch dims for other once we support that use case
+- func: _new_zeros_with_same_feature_meta(Tensor self, Tensor other, int self_num_batch_dims=0) -> Tensor
```

Comment:
Can we make `self_num_batch_dims` kwarg only? We're probably going to add a `other_num_batch_dims` arg in the future. Just in case someone ends up serializing it and gives us FC/BC problems further down the line.
```
_new_zeros_with_same_feature_meta(Tensor self, Tensor other, *, int self_num_batch_dims=0)
```

Reply:
Hmm, I also just realized that we should be skipping the python bindings. It's currently exposed in the torch namespace...
Would we be still worried about serialization FC/BC if there are no Python binding apart from aten.ops?

Perhaps another benefit of making it kwarg only is that we get to bypass the backward compatibility check when we later add the second argument?

Reply:
> Perhaps another benefit of making it kwarg only is that we get to bypass the backward compatibility check when we later add the second argument?

Yup

> Hmm, I also just realized that we should be skipping the python bindings. It's currently exposed in the torch namespace...
Would we be still worried about serialization FC/BC if there are no Python binding apart from aten.ops?

Open question. People are still able to serialize it (via LazyTensor or just seeing it in ` __torch_dispatch__`) but they most likely won't for a while